lib/block_view.go

```diff
@@ -7039,11 +7131,250 @@ func (bav *UtxoView) _connectAcceptNFTBid(
+			return 0, 0, nil, RuleErrorNFTBidLessThanMinBidAmountNanos
+		}
+		if nftEntry.IsBuyNow && txMeta.BidAmountNanos == 0 {
+			return 0, 0, nil, RuleErrorZeroBidOnBuyNowNFT
+		}
+		// Verify that we are not bidding on a Buy Now NFT before the Buy Now NFT Block Height. This should never happen.
+		if nftEntry.IsBuyNow && blockHeight < BuyNowNFTBlockHeight {
+			return 0, 0, nil, errors.Wrapf(RuleErrorBuyNowNFTBeforeBlockHeight, "_connectNFTBid: ")
+		}
+		if nftEntry.IsBuyNow && txMeta.BidAmountNanos >= nftEntry.MinBidAmountNanos {
+			isBuyNowBid = true
+		}
+	}
+
+	// If this is a bid on an NFT that is not "Buy Now" enabled, simply create the bid.
+	if !isBuyNowBid {
+		// Connect basic txn to get the total input and the total output without
+		// considering the transaction metadata.
+		totalInput, totalOutput, utxoOpsForTxn, err := bav._connectBasicTransfer(
+			txn, txHash, blockHeight, verifySignatures)
+		if err != nil {
+			return 0, 0, nil, errors.Wrapf(err, "_connectNFTBid: ")
+		}
+
+		// We assume the tip is right before the block in which this txn is about to be applied.
+		tipHeight := uint32(0)
+		if blockHeight > 0 {
+			tipHeight = blockHeight - 1
+		}
+		// Verify that the transaction creator has sufficient deso to create the bid.
+		spendableBalance, err := bav.GetSpendableDeSoBalanceNanosForPublicKey(txn.PublicKey, tipHeight)
+		if err != nil {
+			return 0, 0, nil, errors.Wrapf(err, "_connectNFTBid: Error getting bidder balance: ")
+		} else if txMeta.BidAmountNanos > spendableBalance && blockHeight > BrokenNFTBidsFixBlockHeight {
+			return 0, 0, nil, RuleErrorInsufficientFundsForNFTBid
+		}
+
+		// Force the input to be non-zero so that we can prevent replay attacks.
+		if totalInput == 0 {
+			return 0, 0, nil, RuleErrorNFTBidRequiresNonZeroInput
+		}
+
+		if verifySignatures {
+			// _connectBasicTransfer has already checked that the transaction is
+			// signed by the top-level public key, which we take to be the poster's
+			// public key.
+		}
+
+		// If an old bid exists, delete it.
+		if prevNFTBidEntry != nil {
+			bav._deleteNFTBidEntryMappings(prevNFTBidEntry)
+		}
+
+		// If the new bid has a non-zero amount, set it.
+		if txMeta.BidAmountNanos != 0 {
+			// Zero bids are not allowed, submitting a zero bid effectively withdraws a prior bid.
+			newBidEntry := &NFTBidEntry{
+				BidderPKID:     bidderPKID.PKID,
+				NFTPostHash:    txMeta.NFTPostHash,
+				SerialNumber:   txMeta.SerialNumber,
+				BidAmountNanos: txMeta.BidAmountNanos,
+			}
+			bav._setNFTBidEntryMappings(newBidEntry)
+		}
+
+		// Add an operation to the list at the end indicating we've connected an NFT bid.
+		utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
+			Type:            OperationTypeNFTBid,
+			PrevNFTBidEntry: prevNFTBidEntry,
+		})
+
+		return totalInput, totalOutput, utxoOpsForTxn, nil
+	} else {
+		newBidEntry := &NFTBidEntry{
```

Comment:
It's sort of weird that we create an NFT bid entry only to immediately delete it later in _helpConnectNFTSold().  I  assumed we would skip the nftBidEntry and just immediately update the nftEntry for the buyer.  I think this is fine though, and certainly works / streamlines the code well.  Maybe just add a quick comment that explains that the bid entry created here is a temporary placeholder and immediately deleted when we call _helpConnectNFTSold()?

Reply:
yep - will add a comment. Basically creating this bid here allows us to reuse the code that is now in _helpConnectNFTBid